pkg/controllers/vrpq/query_reconciler.go

```diff
@@ -71,66 +80,106 @@ func (q *QueryReconciler) Reconcile(ctx context.Context, req *ctrl.Request) (ctr
+func (q *QueryReconciler) collectInfoFromVdb(ctx context.Context) (res ctrl.Result, err error) {
 	vdb := &v1.VerticaDB{}
-	res := ctrl.Result{}
-
-	opts := []restorepointsquery.Option{
-		restorepointsquery.WithInitiator(q.InitiatorPod, q.InitiatorPodIP),
+	if res, err = fetchVDB(ctx, q.VRec, q.Vrpq, vdb); verrors.IsReconcileAborted(res, err) {
+		return res, err
 	}
+	q.Vdb = vdb
 
-	err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
-		var e error
-		if res, e = fetchVDB(ctx, q.VRec, q.Vrpq, vdb); verrors.IsReconcileAborted(res, e) {
-			return e
+	// Build communal storage params if there is not one
+	if q.ConfigurationParams == nil {
+		res, err = q.ConstructConfigParms(ctx)
+		if verrors.IsReconcileAborted(res, err) {
+			return res, err
 		}
-		q.Vdb = vdb
-		// build communal storage params if there is not one
-		if q.ConfigurationParams == nil {
-			res, e = q.ConstructConfigParms(ctx)
-			if verrors.IsReconcileAborted(res, e) {
-				return e
-			}
-		}
-		// extract out the communal and config information to pass down to the vclusterops API.
-		opts = append(opts,
-			restorepointsquery.WithCommunalPath(vdb.GetCommunalPath()),
-			restorepointsquery.WithConfigurationParams(q.ConfigurationParams.GetMap()),
-		)
-
-		return nil
-	})
+	}
 
 	return res, err
 }
 
-// runListRestorePoints will update the status condition before and after calling
-// list restore points api
-// Temporarily, runListRestorePoints will not call the ListRestorePoints API
-// since the dispatcher is not set up yet
-func (q *QueryReconciler) runListRestorePoints(ctx context.Context, _ *ctrl.Request) error {
+// runShowRestorePoints call the vclusterOps API to get the restore points
+func (q *QueryReconciler) runShowRestorePoints(ctx context.Context, dispatcher vadmin.Dispatcher,
+	opts []restorepointsquery.Option) (err error) {
 	// set Querying status condition and state prior to calling vclusterops API
-	err := vrpqstatus.UpdateConditionAndState(ctx, q.VRec.Client, q.VRec.Log, q.Vrpq,
+	err = vrpqstatus.UpdateConditionAndState(ctx, q.VRec.Client, q.VRec.Log, q.Vrpq,
 		v1.MakeCondition(vapi.Querying, metav1.ConditionTrue, "Started"), stateQuerying)
 	if err != nil {
 		return err
 	}
 
-	// API should be called to proceed here
-	// If we receive a failure result from the API, a state message and condition need to be updated
-
-	// clear Querying status condition
-	err = vrpqstatus.UpdateConditionAndState(ctx, q.VRec.Client, q.VRec.Log, q.Vrpq,
-		v1.MakeCondition(vapi.Querying, metav1.ConditionFalse, "Completed"), stateQuerying)
-	if err != nil {
-		return err
+	// call showRestorePoints vcluster API
+	q.VRec.Eventf(q.Vrpq, corev1.EventTypeNormal, events.ShowRestorePointsStarted,
+		"Starting show restore points")
+	start := time.Now()
+	if res, errRun := dispatcher.ShowRestorePoints(ctx, opts...); verrors.IsReconcileAborted(res, errRun) {
```

Comment:
Is there any case where `ShowRestorePoints` will return a `ctrl.Result` with a requeue? We don't do anything if that occurs. I'd like to remove that from `ShowRestorePoints` if we don't have any reason to return one.